collections/ansible_collections/cloudkit/service/roles/cleanup_stale_network_resources/tasks/cleanup.yaml

```diff
@@ -0,0 +1,165 @@
+---
+- name: Get all existing ClusterOrders from specified namespace
+  kubernetes.core.k8s_info:
+    api_version: cloudkit.openshift.io/v1alpha1
+    kind: ClusterOrder
+    namespace: "{{ cleanup_stale_network_resources_namespace }}"
+  register: cluster_orders
+  failed_when: false
+
+- name: Extract cluster names from ClusterOrders
+  ansible.builtin.set_fact:
+    cleanup_stale_network_resources_cluster_orders: "{{ cleanup_stale_network_resources_cluster_orders + [item] }}"
+  loop: "{{ cluster_orders.resources }}"
+  when: cluster_orders.resources is defined
```

Comment:
It's my impression that `cluster_orders.resources` will always be defined; if there are no matching resources, then it'll simply be an empty list. Have you seen something different?

Also, if this is looking for the cluster order names, I think you need something more like:

```
    - name: set cluster order names
      ansible.builtin.set_fact:
        cluster_order_names: "{{ [prefix] | product(cluster_names) | map('join') }}"
      vars:
        prefix: clusterorder_
	    cluster_names: "{{ cluster_orders.resources | map(attribute='metadata.name') }}"
```